### ISSUE TITLE
module 'numba_dpex' has no attribute 'LOCAL_MEM_FENCE'

### ISSUE THREAD

**Issue by zjin-lcf** [IntelPython/numba-dpex#856]
I tried to run the program "kernel_private_memory.py" on Intel DevCloud (2023.0), but got the following error.  Thanks.

numba.core.errors.TypingError: Failed in dpex_nopython mode pipeline (step: nopython frontend)
Internal error at resolving type of attribute "LOCAL_MEM_FENCE" of "$42load_global.17".
module 'numba_dpex' has no attribute 'LOCAL_MEM_FENCE'
During: typing of get attribute at /path/to/examples/kernel_private_memory.py (26)
Enable logging at debug level for details.

File "kernel_private_memory.py", line 26:
    def private_memory_kernel(A):
        <source elided>
        memory[0] = i
        numba_dpex.barrier(numba_dpex.LOCAL_MEM_FENCE)  # local mem fence
        ^

**Comment by mingjie-intel** [comment]
Which version of numba_dpex are you using?

**Comment by zjin-lcf** [comment]
It is part of the oneAPI 2023.0. Could you reproduce the error ?  Thanks

**Comment by mingjie-intel** [comment]
Are you running dpex in Conda environment? If yes, can you share you conda list? I tried on my end with following installed, but can't reproduce the issue. 

```
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    intel
_openmp_mutex             5.1                       1_gnu    intel
bzip2                     1.0.8                hb9a14ef_9    intel
ca-certificates           2022.10.11           h06a4308_0    intel
certifi                   2022.9.24       py310h06a4308_0    intel
dpcpp-cpp-rt              2023.0.0            intel_25370    intel
dpcpp_cpp_rt              2023.0.0            intel_25370    intel
dpctl                     0.14.0          py310h8ea5824_17    intel
dpnp                      0.11.0                   pypi_0    pypi
icc_rt                    2023.0.0            intel_25370    intel
intel-cmplr-lib-rt        2023.0.0            intel_25370    intel
intel-cmplr-lic-rt        2023.0.0            intel_25370    intel
intel-opencl-rt           2023.0.0            intel_25370    intel
intel-openmp              2023.0.0            intel_25370    intel
intelpython               2023.0.0                      1    intel
libffi                    3.3                          14    intel
libgcc-ng                 11.2.0               h1234567_1    intel
libgomp                   11.2.0               h1234567_1    intel
libllvm11                 11.0.0               h3826bc1_1    intel
libstdcxx-ng              11.2.0               h1234567_1    intel
llvm-spirv                11.0.0               h4616538_1    intel
llvmlite                  0.39.1          py310he188f65_0    intel
mkl                       2023.0.0            intel_25398    intel
mkl-dpcpp                 2023.0.0            intel_25398    intel
mkl-service               2.4.0           py310h3605609_14    intel
mkl_fft                   1.3.1           py310hcab1719_22    intel
mkl_random                1.2.2           py310hbf47bc3_22    intel
mkl_umath                 0.1.1           py310hf66a691_32    intel
ncurses                   6.3                  h5eee18b_3    intel
numba                     0.56.3          py310h4e39c7a_2    intel
numba-dpex                0.19.0          py310hfbc08a4_12    intel
numpy                     1.22.3          py310hf0956d0_5    intel
numpy-base                1.22.3          py310h45c9ace_5    intel
openssl                   1.1.1q               h7f8727e_0    intel
packaging                 21.3               pyhd3eb1b0_0    intel
pip                       22.2.2          py310h06a4308_0    intel
pyparsing                 3.0.9           py310h06a4308_0    intel
python                    3.10.8               h3e8fc41_2    intel
readline                  8.2                  h5eee18b_0    intel
setuptools                63.4.1          py310h06a4308_0    intel
six                       1.16.0             pyhd3eb1b0_1    intel
spirv-tools               2020.5               h6bb024c_2    intel
sqlite                    3.40.0               h5082296_0    intel
tbb                       2021.8.0            intel_25334    intel
tbb4py                    2021.8.0        py310_intel_25334    intel
tk                        8.6.12               h1ccaba5_0    intel
wheel                     0.37.1             pyhd3eb1b0_0    intel
xz                        5.2.6                h5eee18b_0    intel
zlib                      1.2.13               h5eee18b_0    intel
```

**Comment by mingjie-intel** [comment]
An update, it is found the version 0.19.0 installed from Intel channel and dppy/label/dev is different. Installing from Intel Channel will solve the issue.

**Comment by diptorupd** [comment]
@zjin-lcf The change to of name from `CLK_LOCAL_MEM_FENCE` to `LOCAL_MEM_FENCE` is coming in the next release. If you want to try out the example from the main branch, please change `LOCAL_MEM_FENCE` to `CLK_LOCAL_MEM_FENCE`. That should work with v0.19.0.

**Comment by diptorupd** [comment]
@mingjie-intel Can you add back `CLK_LOCAL_MEM_FENCE` along with the new flag. We will deprecate the name change in 0.20 and remove in 0.21.

**Comment by diptorupd** [comment]
Fixed in 0.20
